### PR TITLE
private assets signing configuration

### PR DESCRIPTION
**Pull request by Marc-Roig**
Hello! First I do want to say that this is an awesome plugin, and THANK YOU for maintaining it. 

In Strapi we are working on supporting private configurations for providers https://github.com/strapi/strapi/pull/15773 . Meaning that files that are uploaded to be private can be visualised in the Content Manager

This work will be released in 4.9.0.

### What does it do?

This is the work of @jihyukbae, he was trying to test the feature using this specific provider so all props to him! I am just creating this PR and asking for feedback.

This work adds support to visualise gcp private assets in the Content Manager.

We would appreciate if some of you could review this work and express your opinions / suggestions :) This feedback can help reshape and improve the current implementation of file signing.

### Why is it needed?
So users uploading private assets can display those in the content manager.

### How to test it?

- Use 4.9.0 version of strapi.
- Set the `publicFiles` config to false
- Upload a new image in the Media Library. 
- That file url should be signed and you should be able to display that asset.

### CODE REVIEW

**Review by jihyukbae — APPROVED** [review]
Not a maintainer, but I can confirm that this worked for me! @Lith Would you mind taking a look?

**Comment by jihyukbae** [comment]
@Lith Gentle ping for review - this feature would be awesome to have for our team!

**Comment by ciaranwaters** [comment]
Hi. Firstly I'd like to say a big thanks for this project, its brilliant.

Any idea when this will be merged into the main branch and be available to use?

**Comment by ciaranwaters** [comment]
@Marc-Roig It looks like you might need to sign the commits before they can be merged

**Comment by jihyukbae** [comment]
@Marc-Roig Could you sign the commits and merge this PR when you get a chance?

@ciaranwaters In the meantime, if you need/want this feature in advance, you can use [patch-package](https://github.com/ds300/patch-package) and grab the changes in this PR.

**Comment by Marc-Roig** [comment]
@ciaranwaters @jihyukbae 
sorry for the delay, it was a busy week :) 

They should be signed by now, thank you for waiting

**Comment by Marc-Roig** [comment]
I just signed my key, so that's why I imagine it shouws as unverified 🤔  I will wait a bit to see if it changes to Verified, but the emails from my gpg key and this gh account are the same.

**Comment by jihyukbae** [comment]
No worries and thank you Marc!

@Lith Could you approve running the necessary workflows for this PR and merge?

**Comment by kashkashio** [comment]
Hello, for some reason i'm having trouble making this one to work...
Bucket set to fine-grain, bucket access is public. the public flag for the provider is set to false, I updated strapi and plugin to 4.10.1 but still cant make it work. no previews or thumbnails on the strapi admin. any clue? thanks for the great work tho!

**Pull request merged**
